src/modules/Dropdown/Dropdown.js

```diff
@@ -1201,12 +1216,16 @@ export default class Dropdown extends Component {
   }
 
   renderOptions = () => {
-    const { multiple, search, noResultsMessage } = this.props
+    const { maxSelections, multiple, search, noResultsMessage } = this.props
     const { selectedIndex, value } = this.state
     const options = this.getMenuOptions()
 
+    if (!_.isNil(maxSelections) && value.length >= maxSelections) {
+      return <DropdownMessage type={'maxSelections'} value={maxSelections} />
+    }
+
     if (noResultsMessage !== null && search && _.isEmpty(options)) {
-      return <div className='message'>{noResultsMessage}</div>
+      return <DropdownMessage type={'noResultsMessage'} noResultsMessage={noResultsMessage} />
```

Comment:
Let's use factories there, [here is example](https://github.com/Semantic-Org/Semantic-UI-React/blob/v0.77.0/src/modules/Dropdown/DropdownSearchInput.js#L82), we will also use `type` as meanfull prop. This will allow to get following usage:

```jsx
DropdownMessage.create('noResultsMessage')
```

It will allow to use shorthanding options there. We also need to add a new test [like there](https://github.com/Semantic-Org/Semantic-UI-React/blob/v0.77.0/test/specs/collections/Menu/MenuItem-test.js#L12).

Reply:
`noResultsMessage` will become obsolete in `Dropdown`, as it will be breaking change we should show before and after usages in the PR heading.